### ISSUE TITLE
Suggestions/ideas : CarControler

### ISSUE THREAD

**Issue by dad72** [ssatguru/BabylonJS-CharacterController#17]
Hello,

I wonder if it would not be interesting to add a "CarControler" in the same spirit and complementary to the "CharacterControler" . Which would complete this library of controllable objects. Then we could talk about the "FlyControler", but that would be for later. It should be even simpler than the CarControler.
I explain in more detail the idea.

The CarControler : 

- Be able to play an animation to get into a vehicle and attach the character to the seat (parented the character to the seat for example) and detach the character animation.
- The same thing to get out of a vehicle and detach the character and resume the animations of the character.
- Play an animation to turn the steering wheel right and left (Which is enough).
- A camera (The FollowCamera looks interesting here) that follows the vehicle with a position inside (close to the steering wheel) and another in the third person only which is fixed.
- An animation for the wheels eventualy. I saw that animations on the wheel material which was less expensive (may be an idea to explore) This camera does not need to be rotated around the vehicle, but fixed to facilitate driving.
- Sounds for the vehicle that accelerates, brakes, clack-sound, the collision too.
- Moving, moving forward, backward, turning right and left. The movement of the acceleration must be progressive and we can very well imagine a function that would allow this acceleration power to vary depending on the vehicle (if it is heavy like the Transport truck, bus, normal like a standard car, fast , like a Porch, a Ferrari, racing car ...) In short, I find this function very useful to adapt the engine power in a way
- Maybe you have ideas to add...

And without physics (type arcade like GTA doesIf you know this game)
The CharacterControler already has a good basis for making a CarControler. What do you think of the ideas?

I had idea to try to do it, but I have doubts, I think that it should remain in the same spirit as the CharacterControler and that it is to couple, so I ask you if you would be interested in doing it.

I am impatient to know what you think, your reactions, I would be very interested in this and I am sure that on the Babylon forum, many will find it great (I see a lot of little demos with cars)

I can then help you test it and give you feedback, improvements, bugs, help as best as I can. I think you have better programming skills than I do and I don't know much about typescript.

I may be able to help you find a 3D car model that would be used for testing, the final demo, unless you already have a 3D vehicle. Tell me how I can help you, don't hesitate.

Thanks in advance for reading me.

**Comment by ssatguru** [comment]
Yes "CarController" is definitely interesting. I would need something like that for my other project "Vishva". I would like to work on this but right now I do not have much time. What ever time I get goes into "Vishva" or maintaining other projects.  After I have made some decent progress with "Vishva" I will start work on this. Not sure when that will happen though.  
In the meantime checkout https://github.com/swift502/Sketchbook . He has done character controller, car controller and plane controller !!  but it is ThreeJS and physics based  :)

**Comment by dad72** [comment]
Ok. I'll try to do it myself then.
